pinot-connectors/pinot-flink-connector/src/main/java/org/apache/pinot/connector/flink/sink/FlinkSegmentWriter.java

```diff
@@ -63,32 +44,15 @@
  */
 @SuppressWarnings("NullAway")
 @NotThreadSafe
-public class FlinkSegmentWriter implements SegmentWriter {
+public class FlinkSegmentWriter extends FileBasedSegmentWriter {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(FlinkSegmentWriter.class);
-  private static final FileFormat BUFFER_FILE_FORMAT = FileFormat.AVRO;
 
   private final int _indexOfSubtask;
-
-  private TableConfig _tableConfig;
-  private String _tableNameWithType;
-  private BatchIngestionConfig _batchIngestionConfig;
-  private BatchConfig _batchConfig;
-  private String _outputDirURI;
-  private Schema _schema;
-  private Set<String> _fieldsToRead;
-  private RecordTransformer _recordTransformer;
-
-  private File _stagingDir;
-  private File _bufferFile;
   private int _rowCount;
```

Comment:
thanks for cleaning this up!
can rowId and seqId also move to FileBasedSegmentWriter? I think it will benefit from having these fields.

Reply:
rowId at the very least, in case you dont want to move seqId for backward compatibility reasons